e2e-tests/cypress/integration/appellant-submission-decision-letter/appellant-submission-decision-letter.js

```diff
@@ -0,0 +1,32 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Given('user has previously submitted a decision letter file {string}', (filename) => {
+  cy.goToAppellantSubmissionDecisionLetter();
+  cy.uploadDecisionLetterFile(filename);
+  cy.saveAndContinue();
+});
+
+When('user submits a decision letter file {string}', (filename) => {
+  cy.goToAppellantSubmissionDecisionLetter();
+  cy.uploadDecisionLetterFile(filename);
+  cy.saveAndContinue();
+});
+
+Then('user can see that the decision letter file {string} {string} submitted', (filename, submitted) => {
+  if (submitted === 'is') {
+    cy.confirmDecisionLetterFileIsUploaded(filename);
+  } else {
+    cy.confirmDecisionLetterFileIsNotUploaded();
+  }
+});
+
+Then('user is informed that the file is not submitted because {string}', (reason) => {
+  switch (reason) {
+    case 'file type is invalid':
```

Comment:
switch statements are the work of the devil. personal opinion, other opinions are available.

Reply:
What do you suggest here?

Reply:
I'll push this through and discuss separately.